### PR TITLE
Pdi 19886 excel template streaming

### PR DESCRIPTION
**Pull request by tgf**

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 17s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **110**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 8m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **110**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 52s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **110**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 14m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **110**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **112**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by abryant-hv — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 58s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **112**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java [hunk 1]
```diff
@@ -135,17 +134,17 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
 
         // Let's remember where the fields are in the input row
         int outputFieldsCount = meta.getOutputFields().length;
-        data.commentauthorfieldnrs = new int[ outputFieldsCount ];
-        data.commentfieldnrs = new int[ outputFieldsCount ];
-        data.linkfieldnrs = new int[ outputFieldsCount ];
-        data.fieldnrs = new int[ outputFieldsCount ];
+        data.commentauthorfieldnrs = new int[outputFieldsCount];
```
Comment (peterrinehart) [inline plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java:137]: Please check your code style settings; these lines shouldn't have changed.

File: plugins/excel/core/src/test/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterFullTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
+package org.pentaho.di.trans.steps.excelwriter;
+
+import static org.junit.Assert.assertEquals;
```
Comment (peterrinehart) [inline plugins/excel/core/src/test/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterFullTest.java:3]: This file needs a copyright header

File: plugins/excel/core/src/test/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStepTest.java [hunk 3]
```diff
@@ -50,6 +50,8 @@
 
 import java.io.File;
 import java.io.OutputStream;
+import java.net.URL;
```
Comment (peterrinehart) [inline plugins/excel/core/src/test/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStepTest.java:53]: Please update this copyright header

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java (hunk 1 above)
Comment (tgf) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java:137]: code style needs spaces for array brackets?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 12m 49s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **112**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by bconlon-hv — APPROVED** [review]

File: plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java (hunk 1 above)
Comment (abryant-hv) [reply plugins/excel/core/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java]: The array formatting should now be restored.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 14m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl plugins/excel/core
```
---
:ok_hand: All tests passed!
 
Tests run: **112**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9129/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**